src/freertos_drivers/ti/CC32xxSPI.cxx

```diff
@@ -88,10 +97,111 @@ CC32xxSPI::CC32xxSPI(const char *name, unsigned long base, uint32_t interrupt,
+            uint32_t channel_control_options;
+
+            /*
+             * The DMA has a max transfer amount of 1024.  If the transaction is
+             * greater; we must transfer it in chunks.
+             */
+            dmaTransferSize_ = len > MAX_DMA_TRANSFER_AMOUNT ?
+                                                  MAX_DMA_TRANSFER_AMOUNT : len;
+
+            if (tx_buf)
+            {
+                channel_control_options = dmaTxConfig_;
+                buf = (void*)tx_buf;
+            }
+            else
+            {
+                channel_control_options = dmaNullConfig_;
+                buf = &scratch_buffer;
+            }
+
+            /* Setup the TX transfer characteristics & buffers */
+            MAP_uDMAChannelControlSet(dmaChannelIndexTx_ | UDMA_PRI_SELECT,
+                                      channel_control_options);
+            MAP_uDMAChannelAttributeDisable(dmaChannelIndexTx_,
+                                            UDMA_ATTR_ALTSELECT);
+            MAP_uDMAChannelTransferSet(dmaChannelIndexTx_ | UDMA_PRI_SELECT,
+                                       UDMA_MODE_BASIC, buf,
+                                       (void*)(base + MCSPI_O_TX0),
+                                       dmaTransferSize_);
+
+            if (rx_buf)
+            {
+                channel_control_options = dmaRxConfig_;
+                buf = (void*)rx_buf;
+            }
+            else
+            {
+                channel_control_options = dmaNullConfig_;
+                buf = &scratch_buffer;
+            }
+
+            /* Setup the RX transfer characteristics & buffers */
+            MAP_uDMAChannelControlSet(dmaChannelIndexRx_ | UDMA_PRI_SELECT,
+                                      channel_control_options);
+            MAP_uDMAChannelAttributeDisable(dmaChannelIndexRx_,
+                                            UDMA_ATTR_ALTSELECT);
+            MAP_uDMAChannelTransferSet(dmaChannelIndexRx_ | UDMA_PRI_SELECT,
+                                       UDMA_MODE_BASIC,
+                                       (void*)(base + MCSPI_O_RX0),
+                                       buf, dmaTransferSize_);
+
+            /* A lock is needed because we are accessing shared uDMA memory */
+            portENTER_CRITICAL();
+
+            MAP_uDMAChannelAssign(dmaChannelIndexTx_);
+            MAP_uDMAChannelAssign(dmaChannelIndexRx_);
+
+            /* Enable DMA to generate interrupt on SPI peripheral */
+            MAP_SPIDmaEnable(base, SPI_RX_DMA | SPI_TX_DMA);
+            MAP_SPIIntClear(base, SPI_INT_DMARX);
+            MAP_SPIIntEnable(base, SPI_INT_DMARX);
+            MAP_SPIWordCountSet(base, dmaTransferSize_);
+            MAP_IntEnable(interrupt);
+
+            /* Enable channels & start DMA transfers */
+            MAP_uDMAChannelEnable(dmaChannelIndexTx_);
+            MAP_uDMAChannelEnable(dmaChannelIndexRx_);
+
+            portEXIT_CRITICAL();
+
+            MAP_SPIEnable(base);
```

Comment:
why are you toggling the SPI enable bit?
It's especially strange to toggle it outside of locks.

previously we were fine keeping that bit enabled and only using the presence of data in the buffers for clocking.